### ISSUE TITLE
the dimension of label picture?

### ISSUE THREAD

**Issue by maohule** [kyamagu/js-segment-annotator#23]
@kyamagu Hello! 
I have used this tool to make my label picture successful. But when I use this picture to train my  model in caffe ,I get an error which is  'Check failed: outer_num_ * inner_num_ == bottom[1]->count() (250000 vs. 1000000) Number of labels must match number of predictions; e.g., if softmax axis == 1 and prediction shape is (N, C, H, W), label count (number of labels) must be N*H*W, with integer values in {0, 1, ..., C-1}.'.
Then I look at my training details ,I find there are some problems with my label picture. 
I0303 09:36:03.119379  4399 layer_factory.hpp:77] Creating layer data
I0303 09:36:03.119549  4399 net.cpp:86] Creating Layer data
I0303 09:36:03.119555  4399 net.cpp:382] data -> data
I0303 09:36:03.119561  4399 net.cpp:382] data -> label
I0303 09:36:03.131530  4399 net.cpp:124] Setting up data
I0303 09:36:03.131564  4399 net.cpp:131] Top shape: 1 3 500 500 (750000)
**I0303 09:36:03.131567  4399 net.cpp:131] Top shape: 1 1 500 500 4 (1000000)**
I0303 09:36:03.131570  4399 net.cpp:139] Memory required for data: 7000000
I0303 09:36:03.131573  4399 layer_factory.hpp:77] Creating layer data_data_0_split
Why my label picture is "1 1 500 500 4"?What is the meaning of the  '4'?
Look forward to your reply.Thank you!

**Comment by kyamagu** [comment]
@maohule The error depends on how you prepare training data for Caffe, and has nothing to do with js-segment-annotator. I have no idea what `4` indicates here...
